### PR TITLE
SQL-1119: Implement Data Source UI, Kind description, credentials

### PR DESCRIPTION
**Pull request by bucaojit**
UI updates can be seen [here](https://docs.google.com/document/d/1_PtwCWCb5URNRSAaZ_-sOHowdLcGR3UoXIy76Rh3Jdw/edit#bookmark=id.uknko2sh9hsx).  Leaving the SQL Statement field for SQL-1122 because I still need to investigate how to add a field to the `Advanced options` section.

Setting encryption value to to true by default.
Added icons.  
Started on implementing the `Odbc.DataSource()` function

### CODE REVIEW

**Review by terakilobyte — APPROVED** [review]
LGTM. I'm comparing it against some samples Microsoft has, and wondering if there should be more `Config_` [information](https://github.com/microsoft/DataConnectors/blob/master/samples/ODBC/RedshiftODBC/RedshiftODBC.pq#L9-L16)?

**Comment by bucaojit** [comment]
@terakilobyte I'll be adding the Config_* information in SQL-1118 that I've picked up.  Those are options to configure the Datasource such as SqlCapabilities and Extensible options.

File: connector/MongoDBAtlasODBC.pq [hunk 1]
```diff
@@ -1,33 +1,74 @@
...
+    ] & CredentialConnectionString)
+in
+    OdbcDatasource;
+
+GetCredentialConnectionString = () as record =>
     let
-        _message = if (message <> null) then message else "(no message)",
-        a = "Hello from MongoDBAtlasODBC: " & _message
+        Credential = Extension.CurrentCredential(),
+        EncryptionEnabled = Credential[EncryptConnection]? = true,
```
Comment (nbagnard) [inline connector/MongoDBAtlasODBC.pq:39]: Will this conflict with the SSL options users can set in the URI?
https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.ssl

**Review by nbagnard — APPROVED** [review]
LGTM!
I only have 2 comments which are not blocking, more things that we want to think about for the next steps.

**Comment by nbagnard** [comment]
Thank you for adding the screenshot in the doc and adding the links here also :)

File: connector/MongoDBAtlasODBC.pq [hunk 2]
```diff
@@ -1,33 +1,74 @@
...
+    optional options as record)
+    as table meta [
+        Documentation.Name = "MongoDB Atlas SQL"
+    ];
+
+MongoDBAtlasODBCImpl = (mongodbUri as text, database as text, optional options as record) as table => 
+let
+    Uri = ProcessUri(mongodbUri),
+    ConnectionString = [
+        Driver = "ADF_ODBC_DRIVER"
```
[the remark this answers is not in the document]
Comment (bucaojit) [reply connector/MongoDBAtlasODBC.pq:27]: I'll create a ticket to track this.  
Something like "MongoDB Atlas SQL ODBC Driver" would be similar to the examples: [hive](https://github.com/microsoft/DataConnectors/blob/395c34ccbe1ca21675a8244715820a26da8eb622/samples/ODBC/HiveSample/HiveSample.pq#L14),  [Impala](https://github.com/microsoft/DataConnectors/blob/395c34ccbe1ca21675a8244715820a26da8eb622/samples/ODBC/ImpalaODBC/ImpalaODBC.pq#L12)

File: connector/MongoDBAtlasODBC.pq (hunk 1 above)
Comment (bucaojit) [reply connector/MongoDBAtlasODBC.pq:39]: That's a good point.  I will remove the encryption handling here and have it handled in the ProcessUri only so it does not conflict and add a note to validate the encryption is behaving as expected.

**Pull request merged**